packages/visx-axis/src/utils/getTickLabelProps.ts

```diff
@@ -0,0 +1,15 @@
+import { ScaleInput } from '@visx/scale';
+import { TextProps } from '@visx/text';
+import { TickLabelProps, AxisScale } from '../types';
+
+export function getTickLabelProps<Scale extends AxisScale>(
```

Comment:
generally I like the idea of utils to keep things DRY, but in this case I think it makes things a bit harder to read and the types here become complex (e.g., passing in the generics for `Scale` even if it's just an object, or when using this you still have to coerce to `as Partial<TextProps>`) so I wonder if you should just repeat the following logic in the 4x `Axis*` components. wdyt?

```tsx
typeof tickLabelProps === 'function'
  ? tickLabelProps
  : {
    ...defaultTickLabelProps,
    ...tickLabelProps,
  }
```

Reply:
I did think it was quite verbose and hard to read, especially with all the type verbosity. 
I've gone ahead and applied your suggestion in a501cb5d13c18cdfaa1f087d6a170b9345a4e6ee